### PR TITLE
HADOOP-12916

### PR DESCRIPTION
**Pull request by xiaoyuyao**
Create a github PR for code review of v04 patch on the JIRA.

### CODE REVIEW

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java [hunk 1]
```diff
@@ -1636,6 +1640,15 @@ public long getTimeDuration(String name, long defaultValue, TimeUnit unit) {
     return unit.convert(Long.parseLong(vStr), vUnit.unit());
   }
 
+  public long[] getTimeDurations(String name, TimeUnit unit) {
```
Comment (arp7) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java]: Consider returning a List.

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallQueueManager.java [hunk 2]
```diff
@@ -49,25 +57,68 @@
...
   }
 
+  private static <T extends RpcScheduler> T createScheduler(
+      Class<T> theClass, int priorityLevels, String ns, Configuration conf) {
+    // Used for custom, configurable scheduler
+    try {
+      Constructor<T> ctor = theClass.getDeclaredConstructor(int.class,
+          String.class, Configuration.class);
+      return ctor.newInstance(priorityLevels, ns, conf);
+    } catch (RuntimeException e) {
```
Comment (arp7) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallQueueManager.java]: See HDFS-9478 which is fixing exception handling when constructing callqueue instances. We could use a similar fix for createScheduler.

File: hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java (hunk 1 above)
Comment (xiaoyuyao) [inline hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java]: Good point. I will address that in the next patch.

**Comment by elek** [comment]
Can one of the admins verify this patch?

**Comment by aajisaka** [comment]
This issue has been fixed. Closing.

**Pull request closed without merging**